client/keystore/src/lib.rs

```diff
@@ -438,6 +459,29 @@ impl BareCryptoStore for Store {
 	fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> bool {
 		public_keys.iter().all(|(p, t)| self.key_phrase_by_type(&p, *t).is_ok())
 	}
+
+	fn vrf_sign(
+		&self,
+		key_type: KeyTypeId,
+		public: &Sr25519Public,
+		label: &'static [u8],
+		prefix: &'static [u8],
+		randomness: &[u8],
+		slot_number: u64,
+		epoch: u64,
+		threshold: u128,
+	) -> std::result::Result<Option<(Vec<u8>, Vec<u8>)>, TraitError> {
+		let transcript = self.make_vrf_transcript(label, randomness, slot_number, epoch);
```

Comment:
This is leaking BABE consensus code out of the BABE crate. Same with `check_primary_threshold`. Very bad practice IMO